test/models/incomeSourceDetails/incomeSourceIds/IncomeSourceIdHashSpec.scala

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright 2023 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package models.incomeSourceDetails.incomeSourceIds
+
+import models.incomeSourceDetails.incomeSourceIds.IncomeSourceId.mkIncomeSourceId
+import models.incomeSourceDetails.incomeSourceIds.{IncomeSourceId, IncomeSourceIdHash}
+import org.scalatest.Failed
+import testConstants.BaseTestConstants.{testSelfEmploymentId, testSelfEmploymentId2, testSelfEmploymentIdValidation}
+import testUtils.UnitSpec
+
+class IncomeSourceIdHashSpec extends UnitSpec {
+
+  "IncomeSourceIdHash class" should {
+
+    "return IncomeSourceIdHash objects" when {
+
+      "calling both the mkFromIncomeSourceId and mkFromQueryString methods" in {
+        val incomeSourceId: IncomeSourceId = mkIncomeSourceId(testSelfEmploymentId)
+        val incomeSourceIdHashFromIncomeSourceId: IncomeSourceIdHash = IncomeSourceIdHash.mkFromIncomeSourceId(incomeSourceId)
+
+        val incomeSourceIdHashFromQueryString: Either[Throwable, IncomeSourceIdHash] = IncomeSourceIdHash.mkFromQueryString(testSelfEmploymentId)
```

Comment:
we creating IncomeSourceIdHash value out of it string representation, can be something like: "afasv", "asffq" so actual hash code etc